src/farmer.rs

```diff
@@ -675,6 +655,9 @@ impl Config {
             result_receiver: Some(result_receiver),
             node_name,
             base_path,
+            app_info: subspace_farmer::NodeClient::farmer_app_info(&node.rpc_handle)
+                .await
+                .expect("Node is always reachable"),
```

Comment:
Nitpicking:
are we sure that node will always be reachable? maybe user closed node (thinking about substrate cli here).

Reply:
We should be, otherwise it is a bug. If user closed node then we couldn't have `Node` structure (as it is moved in `.close()`)

Reply:
exactly, that's why I thought instead of `expect`, maybe error propagation would be better here. I thought `expect` is for things that we are sure won't happen. 
And, this might happen

Reply:
> are we sure that node will always be reachable

We are actually sure. Even if we are not, types clearly express that it should be reachable. If we panic here, it is a bug, and we do want it to be reported.

> error propagation would be better

Here, some contract on which we rely on is violated, so I think we should just panic here.